src/main/java/it/grid/storm/tape/recalltable/resources/TaskResource.java

```diff
@@ -293,46 +289,59 @@ public Response postNewTask(TaskInsertRequest request) {
 
 		log.info("POST /recalltable/task {}", request);
 
-		validateRequest(request);
+		TaskInsertRequestValidator validator = new TaskInsertRequestValidator(request);
+		if (!validator.validate()) {
+			log.info("BAD REQUEST: {}", validator.getErrorMessage());
+			Response r = Response.status(BAD_REQUEST).entity(validator.getErrorMessage()).build();
+			throw new WebApplicationException(validator.getErrorMessage(), r);
+		}
 
 		StoRI resource = null;
+
 		try {
+
 			resource = service.getResource(request.getStfn());
+
 		} catch (ResourceNotFoundException e) {
-			log.info(e.getMessage());
+
+			log.info(e.getMessage(), e);
 			throw new WebApplicationException(e.getMessage(), e, NOT_FOUND);
+
 		} catch (NamespaceException e) {
-			log.error(e.getMessage());
+
+			log.error(e.getMessage(), e);
 			throw new WebApplicationException(e.getMessage(), e, INTERNAL_SERVER_ERROR);
 		}
 
 		String voName;
+
 		try {
+
 			voName = resource.getVirtualFileSystem()
 					.getApproachableRules()
 					.get(0)
 					.getSubjectRules()
 					.getVONameMatchingRule()
 					.getVOName();
+
 		} catch (NamespaceException e) {
-			log.error(e.getMessage());
+
+			log.error(e.getMessage(), e);
 			throw new WebApplicationException(e.getMessage(), INTERNAL_SERVER_ERROR);
 		}
 
-		if (request.getVoName() != null) {
-			if (!request.getVoName().equals(voName)) {
-				String message = String.format("The voName %s doesn't match the resolved %s",
-						request.getVoName(), voName);
-				log.error(message);
-				throw new WebApplicationException(message, BAD_REQUEST);
-			}
+		if (request.getVoName() != null && !request.getVoName().equals(voName)) {
+			String message = String.format("The voName %s doesn't match the resolved %s",

```

Comment:
Message is a bit cryptic: it should be something like:
"The voName included in the request does not match the voName resolved for this request: %s != %s", request.getVoName(), voName